### ISSUE TITLE
Frontend build fails on ARM64

### ISSUE THREAD

**Issue by ianroberts** [GateNLP/gate-teamware#304]
`npm install` fails when run on linux/arm64 (including in the Docker build on ARM hardware):

```
#27 31.29 > deasync@0.1.21 install /app/frontend/node_modules/deasync
#27 31.29 > node ./build.js
#27 31.29 
#27 31.50 gyp ERR! find Python 
#27 31.50 gyp ERR! find Python Python is not set from command line or npm configuration
#27 31.50 gyp ERR! find Python Python is not set from environment variable PYTHON
#27 31.50 gyp ERR! find Python checking if "python" can be used
#27 31.50 gyp ERR! find Python - "python" is not in PATH or produced an error
#27 31.50 gyp ERR! find Python checking if "python2" can be used
#27 31.50 gyp ERR! find Python - "python2" is not in PATH or produced an error
#27 31.50 gyp ERR! find Python checking if "python3" can be used
#27 31.50 gyp ERR! find Python - "python3" is not in PATH or produced an error
#27 31.50 gyp ERR! find Python 
#27 31.50 gyp ERR! find Python **********************************************************
#27 31.50 gyp ERR! find Python You need to install the latest version of Python.
#27 31.50 gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
#27 31.50 gyp ERR! find Python you can try one of the following options:
#27 31.50 gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
#27 31.50 gyp ERR! find Python   (accepted by both node-gyp and npm)
#27 31.50 gyp ERR! find Python - Set the environment variable PYTHON
#27 31.50 gyp ERR! find Python - Set the npm configuration variable python:
#27 31.50 gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
#27 31.50 gyp ERR! find Python For more information consult the documentation at:
#27 31.50 gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
#27 31.50 gyp ERR! find Python **********************************************************
#27 31.50 gyp ERR! find Python 
#27 31.50 gyp ERR! configure error 
#27 31.50 gyp ERR! stack Error: Could not find any Python installation to use
#27 31.50 gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:307:47)
#27 31.50 gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:136:21)
#27 31.50 gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:179:16)
#27 31.50 gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:271:16)
#27 31.50 gyp ERR! stack     at exithandler (child_process.js:390:5)
#27 31.50 gyp ERR! stack     at ChildProcess.errorhandler (child_process.js:402:5)
#27 31.50 gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
#27 31.50 gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:12)
#27 31.50 gyp ERR! stack     at onErrorNT (internal/child_process.js:472:16)
#27 31.50 gyp ERR! stack     at processTicksAndRejections (internal/process/task_queues.js:82:21)
#27 31.50 gyp ERR! System Linux 5.15.49-linuxkit
#27 31.50 gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
#27 31.50 gyp ERR! cwd /app/frontend/node_modules/deasync
#27 31.50 gyp ERR! node -v v14.21.3
#27 31.50 gyp ERR! node-gyp -v v5.1.1
#27 31.50 gyp ERR! not ok 
#27 31.51 Build failed
```

The root cause of this is that there is no binary build of `deasync` for arm64 - the [suggested fix](https://github.com/vuejs/vue-jest/issues/403#issuecomment-951492051) is to upgrade `@vue/cli-plugin-unit-jest` to a newer version that uses `@vue/vue2-jest` instead of `vue-jest@3.x`, but I'm unclear as to whether this is possible without also upgrading to a newer version of jest and/or nodejs, which may have wider ramifications.

**Comment by ianroberts** [comment]
Why this matters - if we want to be able to run Teamware on AWS, the most cost-effective instances these days are all ARM-based.  I guess it may be possible to build multi-arch Docker images by doing the node build on arm64 and just copying the output files into a final arm64 image but getting it all buildable native on ARM would be better.

**Comment by davidwilby** [comment]
closed by #306
